components/layout_2020/replaced.rs

```diff
@@ -82,25 +118,83 @@ impl ReplacedContent {
             .max_box_size()
             .percentages_relative_to(containing_block);
 
-        let clamp = |inline_size: Length, block_size: Length| {
-            (
-                inline_size.clamp_between_extremums(min_box_size.inline, max_box_size.inline),
-                block_size.clamp_between_extremums(min_box_size.block, max_box_size.block),
-            )
+        let default_object_size = || {
+            // FIXME:
+            // “If 300px is too wide to fit the device, UAs should use the width of
+            //  the largest rectangle that has a 2:1 ratio and fits the device instead.”
+            // “height of the largest rectangle that has a 2:1 ratio, has a height not greater
+            //  than 150px, and has a width not greater than the device width.”
+            physical::Vec2 {
+                x: Length::new(300.),
+                y: Length::new(150.),
+            }
+            .size_to_flow_relative(mode)
+        };
+        let clamp = |inline_size: Length, block_size: Length| Vec2 {
+            inline: inline_size.clamp_between_extremums(min_box_size.inline, max_box_size.inline),
+            block: block_size.clamp_between_extremums(min_box_size.block, max_box_size.block),
```

Comment:
Wondering if we should have a meethod for that, don't we do that for non-replaced things elsewhere in the code?

Reply:
I grepped through the code and didn’t find another instance of computing both at the same time.

In block layout for example, we compute the used width before recurring and the used height after.